gammapy/cube/basic_cube.py

```diff
@@ -0,0 +1,73 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions to perform basic functions for map and cube analysis.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+from ..maps import WcsNDMap
+
+
+def fill_map_counts(event_list, ndmap):
+    """Fill a ``WcsNDMap` with events from an EventList.
+
+     The energy of the events is used for a non-spatial axis homogeneous to energy.
+     The other non-spatial axis names should have an entry in the colum names of the ``EventList``
+
+     Parameters
+     ----------
+     event_list : `~gammapy.data.EventList`
+             the input event list
+     nd_map : `~gammapy.maps.WcsNDMap`
+         Target map
+     """
+    # The list will contain the event table entries to be fed into the WcsNDMap
+    tmp = list()
+    ref_geom = ndmap.geom
+    # Convert events coordinates
+    if ref_geom.coordsys == 'GAL':
+        tmp.append(event_list.galactic)
+    elif ref_geom.coordsys == 'CEL':
+        tmp.append(event_list.radec)
+    else:
+        # should raise an error here. The map is not correct.
+        raise ValueError("Incorrect coordsys of input map.")
+
```

Comment:
There's no need to convert to the coordinate system of the map here.  If you're passing a `SkyCoord` the reference frame will be automatically matched to that of the map.